### PR TITLE
[codex] Rotate mini-game session diagnostics

### PR DESCRIPTION
**Pull request by KirisameYuumio**
## Summary
- keep mini-game diagnostics scoped to the current active scene
- retain only the latest completed/open session for five minutes when no new active scene has started
- drop completed/open sessions as soon as a new active session starts
- expose the retention policy from `/api/game/logs`

## 回归报告 / Regression Report
- Targeted game router pytest: 6 passed, 164 deselected, 2 warnings
- Ruff check on `main_routers/game_router.py`, `utils/game_log.py`, and `tests/unit/test_game_router.py`: passed; existing invalid `# noqa` warnings remain in `main_routers/game_router.py`
- `git diff --check`: passed

## 不拆分理由 / Why Not Split
- This is a single retention-policy fix spanning the in-memory buffer, the retention metadata endpoint, and the matching regression tests.

## Summary by CodeRabbit

* **Bug Fixes**
  * 改进游戏会话调试日志的保留与清理策略：依据 TTL 与保留上限进行过期条目淘汰；当存在活动会话时，已结束日志会被直接丢弃。
  * 更新 `/api/game/logs` 的返回内容：新增保留上限与 TTL 相关字段，移除旧的清理字段。

* **Tests**
  * 重构并细化单元测试，覆盖保留/清理规则、跨分区不串扰以及超 TTL 后的移除断言。

### CODE REVIEW

File: utils/game_log.py [hunk 1]
```diff
@@ -149,8 +160,28 @@ def _get_or_create_game_session_debug_log(
...
+        completed_entries.append(entry)
+
+    completed_entries.sort(
+        key=lambda entry: float(entry.get("ended_at") or entry.get("updated_at") or entry.get("created_at") or 0.0),
+        reverse=True,
+    )
+    for entry in completed_entries[:GAME_SESSION_DEBUG_RETAINED_SESSION_LIMIT]:
+        reference_time = float(entry.get("ended_at") or entry.get("updated_at") or entry.get("created_at") or 0.0)
+        if reference_time and current_time - reference_time <= GAME_SESSION_DEBUG_RETAINED_SESSION_TTL_SECONDS:
+            retained_keys.add(str(entry.get("key") or ""))
```
[the remark this answers is not in the document]
Comment (KirisameYuumio) [reply utils/game_log.py:190]: Fixed in dbabb5bc. Cleanup now drops non-active mini-game debug sessions whenever an active session exists, so a superseded session that ends after the new session starts no longer remains visible during the active scene. Added regression coverage for that handoff path.

File: utils/game_log.py [hunk 2]
```diff
@@ -137,6 +146,8 @@ def _get_or_create_game_session_debug_log(
             "entries": [],
         }
         _game_session_debug_logs[key] = entry
+        if activate:
+            _drop_completed_game_session_debug_logs()
```
[the remark this answers is not in the document]
Comment (KirisameYuumio) [reply utils/game_log.py:155]: Fixed in dbabb5bc. Existing-session activation now also clears completed sessions, matching the new-entry activation path. I also added a regression test for ended-session cleanup while another session is active and added the suggested non-null assertion in the TTL test.

**Comment by KirisameYuumio** [comment]
@codex review

File: utils/game_log.py [hunk 3]
```diff
@@ -145,12 +156,39 @@ def _get_or_create_game_session_debug_log(
...
+    completed_entries: list[dict] = []
+    has_active_session = False
+    for entry in _game_session_debug_logs.values():
+        if entry.get("status") == "active":
+            has_active_session = True
+            continue
+        completed_entries.append(entry)
+
+    if has_active_session:
+        _drop_completed_game_session_debug_logs()
```
[the remark this answers is not in the document]
Comment (KirisameYuumio) [reply utils/game_log.py:180]: Thanks for calling this out. I’m keeping this behavior as-is for this PR because the retention boundary is intentionally the single current mini-game scene for the process, not `lanlan_name` or `game_type`.

Those fields are diagnostic/filter metadata only. A new active scene is expected to clear completed sessions from other characters or game types. Partitioning retention by character would make character identity act as the scene boundary, which does not match future multi-character scenes.

If we later support multiple concurrent mini-game scenes, that should use an explicit scene/scope identifier instead of reusing `lanlan_name` or `game_type`. Commit `c1d5f4c2` documents that intent in code comments and adds regression coverage for the cross-character/cross-type cleanup behavior.

**Review by wehos — APPROVED** [review]
阿巴阿巴
